### DISCUSSION TITLE
Formatting issues in HTML document editing with zed editor！

### DISCUSSION THREAD

**Discussion by sagexiang** [zed-industries/zed#6799] [Feature Requests]
I don’t like automatic formatting when saving an HTML document edited with zed editor. I don’t want to automatically add “/” to meta, img or link tags; the class attribute value “block” is originally at the front but goes to the end after automatic formatting. Went...The format style like Atom is very beautiful! how to solve it? 
Thank you !
<img width="824" alt="截屏2024-01-27 02 26 47" src="https://github.com/zed-industries/zed/assets/50332554/5ec00dc2-9740-41f8-af31-560e86f0efd9">

**Comment by sagexiang** [comment]
I like the formatting style of Atom.

<img width="710" alt="截屏2024-01-27 02 43 09" src="https://github.com/zed-industries/zed/assets/50332554/c17f4a89-1156-46ec-8cbe-c86fa00f27b5">

**Comment by fl0werpowers** [comment]
Seems like Zed uses Prettier by default for formatting html, and addition of trailing slashes in spite of HTML standards is a long-standing issue with Prettier in particular. The workaround for this currently would be [specifying an alternative formatter](https://zed.dev/docs/configuring-zed#formatter) in settings, such as, for example, [js-beautify](https://github.com/beautifier/js-beautify), which you'll have to either install globally or include with your project.

**Comment by waldirbertazzijr** [comment]
Unrelated to @sagexiang question but I got here from a Google search on how to make the available width bigger than 80 on zed while saving with auto format on.

If anyone is looking for this:
```
"language_overrides": {
    "HTML": {
        "prettier": {
            "printWidth": 120
        }
    }
}
```

More option on Prettier API [here](https://prettier.io/docs/en/options).

**Reply by xpe** [reply]
@waldirbertazzijr Where is it documented that using `"prettier"` as a key works as shown above?

**Reply by waldirbertazzijr** [reply]
@xpe I was playing with the autocomplete. With the help of @fl0werpowers comment I figured out that prettier was actually the name of the parser.

<img width="913" alt="Screenshot 2024-07-08 at 20 24 26" src="https://github.com/zed-industries/zed/assets/511680/4ab38a6c-85a8-4ad1-9626-b442a6120d75">

**Reply by nevadascout** [reply]
The `printWidth` option doesn't seem to exist anymore, and adding it to my settings file as suggested here breaks the formatter so it doesn't work at all anymore.

<img width="742" alt="Screenshot 2024-10-23 at 20 36 50" src="https://github.com/user-attachments/assets/ef2244d9-d154-4327-8ace-8108a77ca9b0">

The way to do this now is to set `preferred_line_length` like this:

```
"languages": {
  "HTML": {
    "preferred_line_length": 120
  }
}
```

Alternatively you could set the `max_line_length` in an editorconfig file as this is also supported by Prettier (and will be supported by Zed in the near future [source](https://github.com/zed-industries/zed/issues/19528#issuecomment-2429964820))

**Comment by lukejans** [comment]
@sagexiang The workaround I found was to just not use prettier. I'm using superHTML through the [superHTML zed extension](https://github.com/WeetHet/superhtml-zed?tab=readme-ov-file). This works almost exactly how you are describing you want prettier to work and you can [read more about superHTML here](https://github.com/kristoff-it/superhtml). Here Is the configuration I found that works best (for some reason the configuration on the superHTML README page prevents the VSCode lsp from giving suggestions).

```json
{
  "languages": {
    "HTML": {
      "formatter": { "language_server": { "name": "superhtml" } }
    }
  }
}
```

**Reply by remixer-dec** [reply]
Unfortunately superHTML is completely unconfigurable and in very early stages of development
For my case I ended up with this:
```json
  "languages": {
    "HTML": {
      "prettier": {
        "allowed": false
      },
      "language_servers": ["vscode-html-language-server", "vtsls"]
    }
  }
```

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
